packages/sdk/src/client/backend.ts

```diff
@@ -276,20 +276,36 @@ function processCampaign(backendCampaign: BackendCampaign): Campaign {
         });
     }
 
-    return <Campaign>{
-        ...backendCampaign,
-        chainId: toNumberOrNull(backendCampaign.chainId),
+    const { specification, ...rest } = backendCampaign;
+
+    const campaign: Campaign = {
+        ...rest,
+        chainId: Number(backendCampaign.chainId),
         status,
         pool: {
             ...backendCampaign.pool,
+            // FIXME: we should return the pool chain id from the backend
```

Comment:
About this fixme, the targeted pool will always live in the same chain as the campaign, so it can be removed.

Reply:
Addressed in 2f8d27fdcd5a228a9dd3b57f094e2adfd960ef24.